biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -1005,12 +1094,41 @@ subroutine spawn_patches( currentSite, bc_in)
        
       if ( site_areadis_primary .gt. nearzero) then
           currentPatch               => currentSite%youngest_patch
-          new_patch_primary%older    => currentPatch
-          new_patch_primary%younger  => null()
-          currentPatch%younger       => new_patch_primary
-          currentSite%youngest_patch => new_patch_primary
+          ! insert first primary patch after all the secondary patches, if there are any
+          if (currentPatch%anthro_disturbance_label .eq. secondaryforest ) then
+             foundfirstprimary = .false.
+             do while(associated(currentPatch) .and. .not. foundfirstprimary) 
+                currentPatch => currentPatch%older
+                if (associated(currentPatch)) then
+                   if (currentPatch%anthro_disturbance_label .eq. primaryforest) then
+                      foundfirstprimary = .true.
+                   endif
+                endif
+             end do
+             if (associated(currentPatch)) then
```

Comment:
At this point, 'currentpatch' is the 'firstpatchprimary', right?  
The sense of 'first' isn't necessarily obvious. Could it be 'foundyoungestprimary'?

Reply:
current patch is the youngest patch, could be primary or secondary as far as I can tell.
https://github.com/NGEET/fates/pull/663/files/cbc4ca001f77edf68b6651082d8150fd9e1e40b2#diff-531e45e75218bb11634691970470c90bR1096

Reply:
...but you would have only exited the 'do while' loop if you had -just- found firstpatchprimary, right? 

Unless there are no primary patches?

Reply:
I agree with @rosiealice , also, could you change the comment above to indicate it is looking for the youngest primary patch?  It was confusing trying to figure out the intent as "! insert first primary patch after all secondary patches, if any" was misleading

Reply:
Also, I'm working through what would happen if you only had one patch, and it was secondary...  actually, you provide for that condition below. I see, it would become the oldest patch..  But then again, if there was only 1 patch and it was secondary, we would have no site_areadis_primary, which would mean we are not in this loop anyway.  

I can't find any holes in this logic.

Reply:
ok, changing foundfirstprimary to found_youngest_primary, and also changing/adding to the comments.